packages/pwa-kit-create-app/assets/bootstrap/js/overrides/app/utils/translations.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Dynamically import the translations/messages for a given locale
+ * @param {string} locale
+ * @returns {Promise<Object>} The messages (compiled in AST format) in the given locale.
+ *      If the translation file is not found, return an empty object (so react-intl would fall back to the inline messages)
+ */
+export const fetchTranslations = async (locale) => {
+    const targetLocale =
+        typeof window === 'undefined'
+            ? process.env.USE_PSEUDOLOCALE === 'true'
+                ? 'en-XA'
+                : locale
+            : locale
+
+    let module
+    try {
+        module = await import(`../../../translations/compiled/${targetLocale}.json`)
+    } catch (err) {
+        console.error(err)
+        console.log(
+            'Loading empty messages, so that react-intl would fall back to the inline default messages'
+        )
+        return {}
+    }
+
+    return module.default
+}
```

Comment:
@shethj have we considered whether it's possible to make the dynamic handling part of `packages/template-retail-react-app` and have it handle for both the Extensibility use case **and** the non-Extensibility use case? 

Overriding the base file is exactly the type of thing we want to **help the customer avoid doing** to the greatest extent possible

Reply:
@bfeister Yes that indeed was the goal. But the fact that the translations directory path is actually created at runtime is where the overrides mechanism skips to catch that.

Another solution I tried was to simply extract the path in a different variable (like putting it in a CONTANTS file) and have customers simply override the constants file to replace the path to translations directory. But as-per official webpack documentation: https://webpack.js.org/api/module-methods/#dynamic-expressions-in-import, there's a limitation where the path cannot be fully passed in a variable.

So in our `utils/locale.js` file…

from constants.js
`export const TRANSLATIONS_DIR = '../../../translations'`

This does not work:
`module = await import(`${TRANSLATIONS_DIR}/compiled/${targetLocale}.json`)`

So I ended up extracting just the `fetchTranslations` function to a separate file. @vmarta suggested that function is pretty stable and will not need to be updated frequently and this also allows us to push updates to `utils/locale.js` in the future if needed.

Reply:
I think `await import(something)` doesn't work but I think this works

```
import(<myVarPath>).then(module => { 
  let translationfile = module.default
  // Do stuff 
});
```

Reply:
I'll try that out. But even if that works we still need to have `utils/translations.js` in the generated app because `<myVarPath>` will be built at runtime when the locale is available so we'll not be able to use the overrides mechanism to have the app point to correct translations folder. So this just helps us clean up code in terms of not having to hardcode the path inside the imports.